src/lightning/data/streaming/reader.py

```diff
@@ -47,18 +49,41 @@ def add(self, chunk_indices: List[int]) -> None:
                 if chunk_indice not in self._chunks_index_to_be_processed:
                     self._chunks_index_to_be_processed.append(chunk_indice)
 
+    def delete(self, chunk_indices: List[int]) -> None:
+        """Receive the list of the chunk indices to download for the current epoch."""
+        with self._lock:
+            for chunk_indice in chunk_indices:
+                if chunk_indice not in self._chunks_index_to_be_deleted:
+                    self._chunks_index_to_be_deleted.append(chunk_indice)
+
+    def _delete(self, chunk_index: int) -> None:
+        chunk_filepath, begin, _ = self._config[ChunkedIndex(index=-1, chunk_index=chunk_index)]
+
+        if os.path.exists(chunk_filepath):
+            os.remove(chunk_filepath)
+
     def run(self) -> None:
         while True:
             with self._lock:
+                if len(self._chunks_index_to_be_processed) == 0 and len(self._chunks_index_to_be_deleted) == 0:
+                    sleep(0.0001)
+                    continue
+
+                # Delete the chunks if we are missing disk space.
+                # Check every 10 items to avoid losing too much time on the subprocess
+                if len(self._chunks_index_to_be_deleted) > 10:
+                    if _get_available_amount_of_bytes() < _MINIMUM_DISK_SIZE:
```

Comment:
Could be wrong here, only looked but didn't test myself:
`_get_available_amount_of_bytes` returns KiB because you passed `-k` to the command, yet `MINIMUM_DISK_SIZE` seems to be in bytes. Let's make sure it has the same unit before comparing. I also suggested below to just use `shutil.disk_usage()`, which would make this simpler.

Reply:
Done !